docs/Advanced-Usage/Workloads/FireMarshal-Config.rst

```diff
@@ -0,0 +1,319 @@
+.. _firemarshal-config:
+
+Workload Specification
+=================================
+Workloads are defined by a configuration file and corresponding workload source
+directory, both typically in the ``firesim/sw/firesim-software/workloads/``
+directory. Most paths in the configuration file are assumed to be relative to
+the workload source directory.
+
+Example Configuration File
+-----------------------------
+FireMarshal supports many configuration options (detailed below), many of which
+are not commonly used. We will now walk through an example that uses most of
+the common options: ``workloads/example-fed.json``. In this example, we produce
+a 2-node workload that runs two benchmarks: quicksort and spam-filtering. This
+will require installing a number of packages on Fedora, as well as
+cross-compiling some code. The configuration is as follows:
+
+.. include:: example-fed.json
+  :code: json
+
+The ``name`` field is required and (by convention) should match the name of the
+configuration file. Next is the ``base`` (fedora-base.json). This option
+specifies an existing workload to base off of. FireMarshal will first build
+``fedora-base.json``, and use a copy of its rootfs for example-fed before
+applying the remaining options. Additionally, if fedora-base.json specifies any
+configuration options that we do not include, we will inherit those (e.g. we
+will use the ``linux-config`` option specified by fedora-base). Notice that we
+do not specify a workload source directory. FireMarshal will look in
+``workloads/example-fed/`` for any sources specified in the remaining options.
+
+Next come a few options that specify common setup options used by all jobs in
+this workload. The ``overlay`` option specifies a filesystem overlay to copy
+into our rootfs. In this case, it includes the source code for our benchmarks
+(see ``workloads/example-fed/overlay``). Next is a ``host-init`` option, this
+is a script that should be run on the host before building. In our case, it
+cross-compiles the quicksort benchmark (cross-compilation is much faster than
+natively compiling).
+
+.. include:: example-fed/host-init.sh
+  :code: bash
+
+Next is ``guest-init``, this script should run exactly once natively within our
```

Comment:
Has it been mentioned what's doing hosting during guest-init?

Reply:
It's mentioned in the more verbose description above, but I'll add some clarification to both guest-init and host-init to make it explicit what is meant by 'host' and 'guest'.